src/mono/mono/mini/mini-amd64.c

```diff
@@ -1249,6 +1280,21 @@ mono_arch_set_native_call_context_args (CallContext *ccontext, gpointer frame, M
 		else
 			storage = arg_get_storage (ccontext, ainfo);
 
+#ifdef MONO_ARCH_HAVE_SWIFTCALL
+		if (mono_method_signature_has_ext_callconv (sig, MONO_EXT_CALLCONV_SWIFTCALL)) {
```

Comment:
Do we even need all this code ? Looks like for swift self, we should have something like arg in ireg with the context reg specified in `ainfo->reg`. So arg get storage should already handle it. For swift error similarly the storage is returned, why additional comparison is required ? I think all this code could just be: `if (ainfo == ArgSwiftError) *(gpointer*)storage = 0;`

Reply:
You are right. The SwiftSelf arg is already set properly and should be handled without additional logic. Thanks for the heads up!